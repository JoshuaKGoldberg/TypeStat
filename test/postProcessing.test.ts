import path from "path";
import { expect, test } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

test("Post processing", async () => {
	const caseDir = path.join(import.meta.dirname, "cases/post processing");
	const { actualContent, expectedFilePath, options } =
		await runMutationTest(caseDir);
	await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	expect(options).toMatchSnapshot("options");
}, 50000);
