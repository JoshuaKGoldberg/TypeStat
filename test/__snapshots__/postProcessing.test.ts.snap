// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post processing > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "include": [
      "actual.ts"
    ],
    "compileOnSave": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": false
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": true,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": false,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "include": [
    "actual.ts"
  ],
  "mutators": [
    [
      "fixImportExtensions",
      null
    ],
    [
      "fixIncompleteTypes",
      null
    ],
    [
      "fixMissingProperties",
      null
    ],
    [
      "fixNoImplicitAny",
      null
    ],
    [
      "fixNoImplicitThis",
      null
    ],
    [
      "fixNoInferableTypes",
      null
    ],
    [
      "fixStrictNonNullAssertions",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": [
      [
        "node",
        "./postProcess.cjs"
      ]
    ]
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {}
}"
`;
